folsom/src/main/java/com/spotify/folsom/client/tls/DefaultSSLEngineFactory.java

```diff
@@ -0,0 +1,33 @@
+package com.spotify.folsom.client.tls;
+
+import java.security.NoSuchAlgorithmException;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+
+public class DefaultSSLEngineFactory implements SSLEngineFactory {
```

Comment:
Will this interface need more than this one implementation?

Reply:
Yes, users might want to implement their own way of constructing `SSLEngine`. The most common way is in `DefaultSSLEngineFactory` but this is not the only way.